plugins/guests/debian/cap/configure_networks.rb

```diff
@@ -16,6 +16,17 @@ def self.configure_networks(machine, networks)
             comm.sudo("sed -e '/^#VAGRANT-BEGIN/,$ d' /etc/network/interfaces > /tmp/vagrant-network-interfaces.pre")
             comm.sudo("sed -ne '/^#VAGRANT-END/,$ p' /etc/network/interfaces | tac | sed -e '/^#VAGRANT-END/,$ d' | tac > /tmp/vagrant-network-interfaces.post")
 
+            kernel_ifnames = []
+            comm.sudo("/usr/bin/find /sys/class/net -type l -print | sort") do |type, data|
+              if type == :stdout then
+                data.split("\n").each do |line|
+                  if (line =~ /^\/sys\/class\/net\/(enp[0-9]+s[0-9]+|eth[0-9]+)$/) then
+                    kernel_ifnames << $1
```

Comment:
The name format appears to be described in [systemd's udev-builtin-net_id.c](https://github.com/systemd/systemd/blob/master/src/udev/udev-builtin-net_id.c#L20)